File: testers/avro/java-async/generated-and-checked-in/com/example/events/producer/OrderEventsProducer.java
package com.example.events.producer;

import com.example.events.OrderEvents;
import com.example.events.header.StandardHeaders;
import java.util.concurrent.CompletableFuture;
import org.apache.kafka.clients.producer.Producer;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.clients.producer.RecordMetadata;

/** Type-safe producer for order-events topic */
public record OrderEventsProducer(Producer<String, OrderEvents> producer, String topic)
    implements AutoCloseable {
  public OrderEventsProducer(Producer<String, OrderEvents> producer) {
    this(producer, "order-events");
  }

  public OrderEventsProducer withProducer(Producer<String, OrderEvents> producer) {
    return new OrderEventsProducer(producer, topic);
  }

  public OrderEventsProducer withTopic(String topic) {
    return new OrderEventsProducer(producer, topic);
  }

  /** Close the producer */
  @Override
  public void close() {
    producer.close();
  }

  /** Send a message to the topic asynchronously */
  public CompletableFuture<RecordMetadata> send(String key, OrderEvents value) {
    return CompletableFuture.supplyAsync(
            () -> {
              CompletableFuture<RecordMetadata> future = new CompletableFuture<>();
              producer.send(
                  new ProducerRecord<String, OrderEvents>(topic, key, value),
                  (result, exception) -> {
                    if (exception != null) {
                      future.completeExceptionally(exception);
                    } else {
                      future.complete(result);
                    }
                  });
              return future;
            })
        .thenCompose(f -> f);
  }

  /** Send a message with headers to the topic asynchronously */
  public CompletableFuture<RecordMetadata> send(
      String key, OrderEvents value, StandardHeaders headers) {
    return CompletableFuture.supplyAsync(
            () -> {
              CompletableFuture<RecordMetadata> future = new CompletableFuture<>();
              producer.send(
                  new ProducerRecord<String, OrderEvents>(
                      topic, null, key, value, headers.toHeaders()),
                  (result, exception) -> {
                    if (exception != null) {
                      future.completeExceptionally(exception);
                    } else {
                      future.complete(result);
                    }
                  });
              return future;
            })
        .thenCompose(f -> f);
  }
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/producer/OrderEventsProducer.kt
package com.example.events.producer

import com.example.events.OrderEvents
import com.example.events.header.StandardHeaders
import java.io.Closeable
import java.util.concurrent.Future
import org.apache.kafka.clients.producer.Producer
import org.apache.kafka.clients.producer.ProducerRecord
import org.apache.kafka.clients.producer.RecordMetadata

/** Type-safe producer for order-events topic */
data class OrderEventsProducer(
  val producer: Producer<kotlin.String, OrderEvents>,
  val topic: kotlin.String = "order-events"
) : Closeable {
  /** Close the producer */
  override fun close() {
    producer.close()
  }

  /** Send a message to the topic */
  fun send(
    key: kotlin.String,
    value: OrderEvents
  ): Future<RecordMetadata> {
    return producer.send(ProducerRecord<kotlin.String, OrderEvents>(topic, key, value))
  }

  /** Send a message with headers to the topic */
  fun send(
    key: kotlin.String,
    value: OrderEvents,
    headers: StandardHeaders
  ): Future<RecordMetadata> {
    return producer.send(ProducerRecord<kotlin.String, OrderEvents>(topic, null, key, value, headers.toHeaders()))
  }
}